src/backend/distributed/planner/distributed_planner.c

```diff
@@ -593,6 +566,253 @@ CreateDistributedPlan(PlannedStmt *localPlan, Query *originalQuery, Query *query
 }
 
 
+/*
+ * CreateDistributedSelectPlan generates a distributed plan for a SELECT query.
+ * It goes through 3 steps:
+ *
+ * 1. Try router planner
+ * 2. Generate subplans for CTEs and complex subqueries
+ *    - If any, go back to step 1 by calling itself recursively
+ * 3. Logical planner
+ */
+static DistributedPlan *
+CreateDistributedSelectPlan(uint64 planId, Query *originalQuery, Query *query,
+							ParamListInfo boundParams, bool hasUnresolvedParams,
+							PlannerRestrictionContext *plannerRestrictionContext)
+{
+	DistributedPlan *distributedPlan = NULL;
+	MultiTreeRoot *logicalPlan = NULL;
+	DeferredErrorMessage *error = NULL;
+	List *subPlanList = NIL;
+
+	/*
+	 * For select queries we, if router executor is enabled, first try to
+	 * plan the query as a router query. If not supported, otherwise try
+	 * the full blown plan/optimize/physical planing process needed to
+	 * produce distributed query plans.
+	 */
+	if (EnableRouterExecution)
```

Comment:
I just noticed `MultiRouterPlannableQuery` also checks this GUC, which is a bit odd since `CreateRouterPlan` asserts that it is true before calling `MultiRouterPlannableQuery`. Anyway, it seems prettier to me if we remove this check and the Assert and just let it fall through to `MultiRouterPlannableQuery`.